tests/bluetooth/bsim_bt/bsim_test_eatt_notif/src/server_test.c

```diff
@@ -105,7 +105,7 @@ static uint8_t discover_func(struct bt_conn *conn,
 
 static void gatt_discover(void)
 {
-	struct bt_gatt_discover_params discover_params;
+	static struct bt_gatt_discover_params discover_params;
```

Comment:
what is the rationale behind this change?

Reply:
- It zeroes out the struct, preventing uninitialized fields
- GATT param structs need to be valid until the response is received

It is not strictly needed in this case because the fields are explicitly set and we wait for the response in the function so an auto-variable is fine.

Reply:
> It zeroes out the struct, preventing uninitialized fields

that should be done with `struct bt_gatt_discover_params discover_params = { 0 };`

> GATT param structs need to be valid until the response is received

So we have a dangling pointer?

Reply:
> > GATT param structs need to be valid until the response is received
>
> So we have a dangling pointer?

The param struct is a parameter in the response callback, and it is used in the stack to figure out what the response callback is and call it. The param struct must therefore be valid until the response callback is called. If it is a stack allocated variable that goes out of scope before the response callback is called, yes we have a dangling pointer. In this case that won't happen because `gatt_discover()` waits for a flag that is set in the response callback.

Reply:
So `static` can be removed in this case and it won't change anything, but in general GATT param structs should be static.

Reply:
If it does not need to be static, I would prefer it to not be. As you point out, static lifetime is not needed in this case. 

The lifetime requirement as documented on `bt_gatt_discover` follows.

> @p params must remain valid until start of callback where iter `attr` is `NULL` or callback will return `BT_GATT_ITER_STOP`.